lib/adapters/http_server.js

```diff
@@ -103,21 +137,25 @@ var Read = source.extend({
 
         this.trigger('error', err);
         res.end();
-    },
-
-    teardown: function() {
-        if (this.server) {
-            this.server.close();
-        }
+    }
 
-        this.emit_eof();
+    teardown() {
+        return new Promise((resolve, reject) => {
+            if (this.server) {
+                this.server.close(() => {
```

Comment:
Except that I don't want to pass through any arguments that come back in the callback as my resolution of teardown.

Not that it really matters but I think the less-pithy two extra lines of code improve clarity.